service/controller/resource/promxy/servergroup.go

```diff
@@ -0,0 +1,172 @@
+	// NaN -- the problem being that for range vectors they filter out all "stale" samples
+	// meaning that it isn't possible to get a "raw" dump of data through the query/query_range v1 API
+	// The only option that exists in reality is the "remote read" API -- which suffers
+	// from the same memory-balooning problems that the HTTP+JSON API originally had.
+	// It has **less** of a problem (its 2x memory instead of 14x) so it is a viable option.
+	RemoteRead bool `yaml:"remote_read"`
+	// RemoteReadPath sets the remote read path for the hosts in this servergroup
+	RemoteReadPath string `yaml:"remote_read_path"`
+	// HTTP client config for promxy to use when connecting to the various server_groups
+	// this is the same config as prometheus
+	HTTPConfig HTTPClientConfig `yaml:"http_client"`
+	// Scheme defines how promxy talks to this server group (http, https, etc.)
+	Scheme string `yaml:"scheme"`
+	// Labels is a set of labels that will be added to all metrics retrieved
+	// from this server group
+	Labels model.LabelSet `yaml:"labels"`
+	// RelabelConfigs are similar in function and identical in configuration as prometheus'
+	// relabel config for scrape jobs. The difference here being that the source labels
+	// you can pull from are from the downstream servergroup target and the labels you are
+	// relabeling are that of the timeseries being returned. This allows you to mutate the
+	// labelsets returned by that target at runtime.
+	// To further illustrate the difference we'll look at an example:
+	//
+	//      relabel_configs:
+	//    - source_labels: [__meta_consul_tags]
+	//      regex: '.*,prod,.*'
+	//      action: keep
+	//    - source_labels: [__meta_consul_dc]
+	//      regex: '.+'
+	//      action: replace
+	//      target_label: datacenter
+	//
+	// If we saw this in a scrape-config we would expect:
+	//   (1) the scrape would only target hosts with a prod consul label
+	//   (2) it would add a label to all returned series of datacenter with the value set to whatever the value of __meat_consul_dc was.
+	//
+	// If we saw this same config in promxy (pointing at prometheus hosts instead of some exporter), we'd expect a similar behavior:
+	//   (1) only targets with the prod consul label would be included in the servergroup
+	//   (2) it would add a label to all returned series of this servergroup of datacenter with the value set to whatever the value of __meat_consul_dc was.
+	//
+	// So in reality its "the same", the difference is in prometheus these apply to the labels/targets of a scrape job,
+	// in promxy they apply to the prometheus hosts in the servergroup - but the behavior is the same.
+	RelabelConfigs []*relabel.Config `yaml:"relabel_configs,omitempty"`
+	// Hosts is a set of discovery.Config options that allow promxy to discover
+	// all hosts in the server_group
+	KubernetesSDConfigs []*kubernetes.SDConfig `yaml:"kubernetes_sd_configs,omitempty"`
```

Comment:
In the original jacksontj/promxy code, the used type is *discovery.Config[] which cannot be deserialized as Config is an interface and this needs to be inlinded. Maybe you have an idea on how to configure it back to *discovery.Config[] yaml:",inline"

Reply:
It seems to be `"github.com/prometheus/prometheus/discovery/config".ServiceDiscoveryConfig` here https://github.com/jacksontj/promxy/blob/master/pkg/servergroup/config.go#L86, or am I looking at a wrong place?

Reply:
So, yes, in prometheus 1.8 code which translates to https://github.com/prometheus/prometheus/blob/924ce978b734e8a738ef28e2207997a34710cfa6/discovery/discovery.go#L58 in prom 2.20 code